okhttp/src/main/java/okhttp3/OkHttpClient.kt

```diff
@@ -785,6 +785,8 @@ open class OkHttpClient internal constructor(
      *    }
      * ```
      *
+     * n.b. OkHttp will generate a wrapper with this method for you, if you don't.
```

Comment:
Might remove this, it seems problematic.  We either use TrustManagerImpl or we fake with emptyList()

Reply:
Generally we need this because the Default TrustManager on Android is TrustManagerImpl.  No one else should implement that logic.  So if you use overrides we currently have two choices

1) You wrap an existing TrustManagerImpl, and delegate to that.  We can define an optional typed interface to extend that allows for that.
2) You don't e.g. InsecureTrustManager - in which case we shouldn't ask people to implement it, we should just return an emptyList() ourselves.

But we should make it simpler to get right.

Reply:
OkHttpTrustManagerJvm does something vaguely similar, if you do implement X509ExtendedTrustManager we use that, otherwise we drop the socket param and call the check check method.